packages/@o3r/core/src/store/async/async.operators.ts

```diff
@@ -14,10 +14,10 @@ import { AsyncRequest, ExtractFromApiActionPayloadType, FromApiActionPayload } f
  * @param cancelRequestActionFactory function that returns the action to emit in case the FromApi action is 'cancelled' because a new action was received by the switchMap
  */
 // eslint-disable-next-line @typescript-eslint/ban-types
-export function fromApiEffectSwitchMap<T extends FromApiActionPayload<any>, S extends ExtractFromApiActionPayloadType<T>, U extends Action, V extends Action, W extends Action>(
+export function fromApiEffectSwitchMap<T extends FromApiActionPayload<any>, S extends ExtractFromApiActionPayloadType<T>, U extends Action, V extends Action | never, W extends Action | never>(
   successHandler: (result: S, action: T) => U | Observable<U>,
```

Comment:
We could also have an optional successHandler, in case we want to do an effect reacting to only errors for eg

Reply:
Unfortunately, this is not possible because the success callback is required to determine the operator return type